### DISCUSSION TITLE
Thoughts on having <form.Provider> behave as form <form>

### DISCUSSION THREAD

**Discussion by linhub15** [TanStack/form#575] [Q&A] [answered]
The current way is like this

```jsx
<form.Provider>
  <form onSubmit={...}>
    {...}
  </form>
<form.Provider>
```

Would it be a bad idea to do this?

```jsx
<form.Provider as="form" onSubmit={...}>
    {...}
<form.Provider>
```

**Comment by crutchcorn** [comment, marked as answer]
There's a lot of challenges with doing things this way. It means we'd need to allow form props to be passed, it wouldn't be headless, and more.
